### PR TITLE
docs/fluentui_development_notes

### PR DESCRIPTION
**Pull request by rymeskar**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Adding development notes for fluentui

#### Focus areas to test

(optional)

### CODE REVIEW

**Review by pompomon — APPROVED** [review]

File: packages/fluentui/CONTRIBUTING.md [hunk 1]
```diff
@@ -140,6 +140,43 @@ Please always use [`lerna add`][16] to manage all dependencies including interna
...
+
+### Collection of Little Ideas
+* Changelog has a special format.
+* Always try to find the most relevant and similar component to get inspired by.
+* Try to extract [pure functions](https://en.wikipedia.org/wiki/Pure_function) outside of component's body; possibly into special files.
+* When developing date-time-utilities, one must run `yarn change` in the root and then  `yarn update-api` in the date-time-utilities folder.
+* Don't name boolean flags with 'is' prefix.
+* `Yarn build` is not necessary when working with FluentUI. You should just call `yarn` (resolve packages) and then `yarn start` (start application).
+* If you have issues with CI there is a [channel](https://teams.microsoft.com/l/channel/19%3a5b644df17ad64b89a6b7947969382baa%40thread.skype/Engineering%2520Systems?groupId=64c2bd5c-33c2-4898-918f-4a02a6ac98e1&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) to report such issues.
+* For benchmarking javaScript, one can use the https://jsperf.com .
```
Comment (pompomon) [inline packages/fluentui/CONTRIBUTING.md:154]:
```suggestion
* For benchmarking JavaScript, one can use the https://jsperf.com .
```
_nit_

File: packages/fluentui/CONTRIBUTING.md [hunk 2]
```diff
@@ -140,6 +140,43 @@ Please always use [`lerna add`][16] to manage all dependencies including interna
...
+* Testing
+  * For majority of components, visual test in combination with `isConformant` test is enough.
+  * Visual tests are based off examples. You can extend the examples by also providing 'steps' action tests.
+* Props
+  * Props should be a flattened object.
+  * Props should take advantage of optional properties in combination with defaultValues.
+* Event Handlers
+  * Going forward, we should use `onChange?: (e: React.SOME_EVENT<EL_TYPE>, data: DatepickerProps & { value: IDay }>)`. The currently used ComponentEventHandler uses React.SyntheticEvent and is able to reason about the proper type then.
+  * We use lodash's `_.invoke()`, the function ensures that the prop actually exists.
+  * We should work carefully with `event.preventDefault()`. Only invoke in necessary cases ([Example]( https://github.com/Semantic-Org/Semantic-UI-React/pull/4041)).
```
Comment (pompomon) [inline packages/fluentui/CONTRIBUTING.md:171]: Please could you add a case from Fluent UI? ;)

Comment (rymeskar) [reply packages/fluentui/CONTRIBUTING.md:171]: Fixed.

File: packages/fluentui/CONTRIBUTING.md (hunk 1 above)
Comment (rymeskar) [reply packages/fluentui/CONTRIBUTING.md:154]: Fixed

**Review by pompomon — APPROVED** [review]

**Pull request merged**